x/lightclient/keeper/hook_listener.go

```diff
@@ -45,6 +45,8 @@ func (hook rollappHook) AfterUpdateState(
 		if ok {
 			hook.k.SetCanonicalClient(ctx, rollappId, client)
 		}
+	}
+	if !ok {
 		return nil
 	}
```

Comment:
this avoid an incorrect early return which caused deleting the header signer to be skipped